src/hooks/useEscrowContractUsers.ts

```diff
@@ -0,0 +1,48 @@
+import { createOwner, TitleEscrowOwner } from "@govtechsg/oa-token";
+
+import { getLogger } from "../utils/logger";
+
+import { useEffect, useState } from "react";
+import { useEthereumTransactionState, TransactionStateStatus, StateType } from "./useEthereumTransactionState";
+import { useWeb3Provider } from "./useWeb3Provider";
+const { trace, error } = getLogger("hooks:useEscrowContractUsers");
+
+type UseEscrowContractUsers = { state: StateType; holderAddress: string; beneficiaryAddress: string };
+
+export const useEscrowContractUsers = ({
+  escrowContractAddress
+}: {
+  escrowContractAddress: string;
+}): UseEscrowContractUsers => {
+  trace(`escrow contract address ${escrowContractAddress}`);
+  const [holderAddress, setHolderAddress] = useState("");
+  const [beneficiaryAddress, setBeneficiaryAddress] = useState("");
+  const { web3Provider } = useWeb3Provider();
+  const [state, dispatch] = useEthereumTransactionState();
+
+  useEffect(() => {
+    async function fetchEscrowContractUsers() {
+      try {
+        dispatch({ type: TransactionStateStatus.LOADING });
+        const titleEscrowInstance = (await createOwner({
+          address: escrowContractAddress,
+          web3Provider
+        })) as TitleEscrowOwner;
```

Comment:
why the need to cast ?

Reply:
instance returned from createowner method are TitleescrowOwner or EthereumAccountOnwer. EthereumAccountOwner doesn't have the methods using further, which is causing the error in type checks.

Reply:
https://github.com/Open-Attestation/oa-token/blob/7cdb78ca5909fce77cb579c625191150d4b688c0/src/owner.ts#L36

https://github.com/Open-Attestation/oa-token/blob/7cdb78ca5909fce77cb579c625191150d4b688c0/src/owner.ts#L40

all classes that extends Owner have to implement it

Reply:
const [beneficiary, holder] = await Promise.all([
          titleEscrowInstance.beneficiary(),
          titleEscrowInstance.holder()
        ]);

These methods are not there on EthereumAccountOwner which is causing the problem .

Reply:
ok i've added a type guard on isTitleEscrow() to fix this
https://github.com/Open-Attestation/oa-token/pull/32
can approve and merge it if ok

Reply:
updated oa-token and removed this type cast.